operator/src/runners/fixtures/membership.rs

```diff
@@ -42,33 +46,131 @@ pub struct SP1ICS07MembershipFixture {
 pub async fn run(args: MembershipCmd) -> anyhow::Result<()> {
-    assert!(!args.key_paths.is_empty());
+    assert!(!args.membership.key_paths.is_empty());
 
     let tm_rpc_client = HttpClient::from_env();
-    let verify_mem_prover = SP1ICS07TendermintProver::<MembershipProgram>::default();
 
     let trusted_light_block = tm_rpc_client
-        .get_light_block(Some(args.trusted_block))
+        .get_light_block(Some(args.membership.trusted_block))
         .await?;
 
     let genesis = SP1ICS07TendermintGenesis::from_env(
         &trusted_light_block,
-        args.trust_options.trusting_period,
-        args.trust_options.trust_level,
+        args.membership.trust_options.trusting_period,
+        args.membership.trust_options.trust_level,
     )
     .await?;
 
     let trusted_client_state = ClientState::abi_decode(&genesis.trusted_client_state, false)?;
     let trusted_consensus_state =
         SolConsensusState::abi_decode(&genesis.trusted_consensus_state, false)?;
+
+    let membership_proof = if args.union {
+        run_union_membership(
+            &tm_rpc_client,
+            args.membership.base64,
+            args.membership.key_paths,
+            args.membership.trusted_block,
+            trusted_consensus_state,
+        )
+        .await?
+    } else {
+        run_sp1_membership(
+            &tm_rpc_client,
+            args.membership.base64,
+            args.membership.key_paths,
+            args.membership.trusted_block,
+            trusted_consensus_state,
+        )
+        .await?
+    };
+
+    let fixture = SP1ICS07MembershipFixture {
+        genesis,
+        proof_height: trusted_client_state.latestHeight.abi_encode(),
+        membership_proof: membership_proof.abi_encode(),
+    };
+
+    match args.membership.output_path {
+        OutputPath::File(path) => {
+            // Save the proof data to the file path.
+            std::fs::write(PathBuf::from(path), serde_json::to_string_pretty(&fixture)?).unwrap();
+        }
+        OutputPath::Stdout => {
+            println!("{}", serde_json::to_string_pretty(&fixture)?);
+        }
+    }
+
+    Ok(())
+}
+
+/// Generates a union membership proof for the given args
+#[allow(
+    clippy::missing_errors_doc,
+    clippy::missing_panics_doc,
+    clippy::module_name_repetitions
+)]
+pub async fn run_union_membership(
+    tm_rpc_client: &HttpClient,
+    is_base64: bool,
+    key_paths: Vec<String>,
+    trusted_block: u32,
+    trusted_consensus_state: SolConsensusState,
+) -> anyhow::Result<MembershipProof> {
+    assert!(
+        key_paths.len() == 1,
+        "Union membership only supports one key-value pair"
+    );
+
+    let path: Vec<Vec<u8>> = if is_base64 {
+        key_paths[0]
+            .split('/')
+            .map(subtle_encoding::base64::decode)
+            .collect::<Result<_, _>>()?
+    } else {
+        vec![b"ibc".into(), key_paths[0].as_bytes().to_vec()]
+    };
+    assert_eq!(path.len(), 2);
```

Comment:
```suggestion
    // the program could support longer key paths, but the operator only supports 2
    // because the current assumption is that the Cosmos SDK will always have 2
    assert_eq!(path.len(), 2);
```